### PR TITLE
AudioVideo: fixes

### PR DESCRIPTION
**Pull request by cchampet**
* Clean AudioVideo + fixes
* Need to review especially last commit: ProcessGraph, call begin/endSequence foreach plugin only one time (fix issue #399).

### CODE REVIEW

File: libraries/tuttle/src/tuttle/host/graph/ProcessGraph.cpp [hunk 1]
```diff
@@ -607,14 +607,24 @@ bool ProcessGraph::process( memory::IMemoryCache& outCache )
 
 	TUTTLE_LOG_INFO( "[Process render] start" );
 
-	//--- RENDER
-	// at each frame
-	BOOST_FOREACH( const TimeRange& timeRange, timeRanges )
+	// Begin range of frames
+	TimeRange beginTimeRange( timeRanges.back() );
```
Comment (fabiencastan) [inline libraries/tuttle/src/tuttle/host/graph/ProcessGraph.cpp]: rename as "globalTimeRange"

File: plugins/image/io/AudioVideo/src/common/util.cpp [hunk 2]
```diff
@@ -68,147 +64,121 @@ LibAVParams::~LibAVParams()
...
 
-void LibAVParams::fetchLibAVParams( OFX::ImageEffect& plugin, avtranscoder::OptionArrayMap& optionArrayMap, const std::string& prefix )
+void LibAVParams::fetchLibAVParams( OFX::ImageEffect& plugin, const std::string& prefix )
 {
 	// iterate on map keys
-	BOOST_FOREACH( avtranscoder::OptionArrayMap::value_type& subGroupOption, optionArrayMap )
+	BOOST_FOREACH( avtranscoder::OptionArrayMap::value_type& subGroupOption, _avOptions )
 	{
-		const std::string subGroupName = subGroupOption.first;
+		const std::string detailledName = subGroupOption.first;
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/common/util.cpp]: spelling: "detailedName"

Comment (cchampet) [inline plugins/image/io/AudioVideo/src/common/util.cpp]: Done here 3c07ace05d0a0b17ae0f9e8b85f2d200a622a042

File: plugins/image/io/AudioVideo/src/common/util.cpp [hunk 3]
```diff
@@ -17,50 +18,45 @@ namespace plugin {
 namespace av {
 namespace common {
 
-LibAVParams::LibAVParams( const std::string& prefixScope, const std::string& prefixOperation )
+LibAVParams::LibAVParams( const std::string& prefixScope, const int flags, const bool isDetailled )
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/common/util.cpp]: isDetailed

File: plugins/image/io/AudioVideo/src/common/util.cpp [hunk 4]
```diff
@@ -349,64 +339,65 @@ void LibAVParams::setOption( const std::string& libAVOptionName, const std::stri
...
 		if( paramChoice )
 		{
 			paramChoice->setValue( option.getInt() );
-			return;
+			return true;
 		}
 	}
 	catch( std::exception& e )
 	{
 		TUTTLE_TLOG( TUTTLE_INFO, "Can't set option " << libAVOptionName << " to " << value << ": " << e.what() );
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/common/util.cpp]: TUTTLE_LOG_WARNING ?

File: plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp [hunk 5]
```diff
@@ -1046,30 +993,25 @@ void AVWriterPlugin::beginSequenceRender( const OFX::BeginSequenceRenderArgument
...
-		avtranscoder::Logger::setLogLevel( AV_LOG_QUIET );
 }
 
-/**
- * @brief The overridden render function
- * @param[in]   args     Rendering parameters
- */
 void AVWriterPlugin::render( const OFX::RenderArguments& args )
 {
 	if( _clipSrc->fetchImage( args.time ) && _clipDst->fetchImage( args.time ) )
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp]:
```
if( !_clipSrc->fetchImage( args.time ) || !_clipDst->fetchImage( args.time ) )
    return;
```

File: plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp [hunk 6]
```diff
@@ -1175,6 +1135,96 @@ std::string AVWriterPlugin::getSampleFormatName( const std::string& audioCodecNa
...
+		{
+			const size_t indexCodec = std::distance( codecs.begin(), itCodec );
+			if( indexCodec < codecs.size() )
+				_paramAudioCodec->setValue( indexCodec );
+			break;
+		}
+	}
+}
+
+void AVWriterPlugin::cleanProfile( avtranscoder::ProfileLoader::Profile& profileToClean, const std::string& prefix )
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp]: doc
What does this function?

File: plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.hpp [hunk 7]
```diff
@@ -96,15 +93,28 @@ class AVWriterPlugin : public WriterPlugin
...
 	//@}
 
+	//@{
+	void setFormatParam( const std::string& formatShortName );
+	void setVideoCodecParam( const std::string& videoCodecShortName );
+	void setAudioCodecParam( const std::string& audioCodecShortName );
+	//@}
+
+	// remove some bad values for wrapper/encoder
+	// insert value from preset file which had not taken by an OFX parameter
```
Comment (fabiencastan) [inline plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.hpp]: Ok, it's documented here, but the message is not clear to me.
Need to use doxygen style.

File: plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp (hunk 6 above)
Comment (cchampet) [inline plugins/image/io/AudioVideo/src/writer/AVWriterPlugin.cpp]: Done here 4eb6a2d38ef7a29081a8ef8551968a98f56f75e8

**Pull request merged**